api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java

```diff
@@ -666,9 +676,189 @@ public void shortenedStackTrace_shouldReturnNullIfStackTraceIsNull() {
 	}
 	
 	/**
-	 * @see OpenmrsUtil#nullSafeEqualsIgnoreCase(String,String)
+	 * @see OpenmrsUtil#folderContains(File, String)
+	 */
+	@Test
+	public void folderContains_shouldReturnTrueWhenFolderContainsFileWithGivenName() {
+		String path = System.getProperty("user.dir");
+		String separator = FileSystems.getDefault().getSeparator();
+		File folder = new File(path + separator + "folder");
+		folder.mkdir();
+		File file = new File(path + separator + "folder" + separator + "file");
+		try {
+			file.createNewFile();
+			assertTrue(OpenmrsUtil.folderContains(folder, "file"));
+			file.delete();
+
+		} 
+		catch (Exception e) {
+		}
+		folder.delete();
+	}
+	
+	/**
+	 * @see OpenmrsUtil#folderContains(File, String)
+	 */
+	@Test
+	public void folderContains_shouldReturnFalseWhenFolderDoesNotContainFileWithGivenName() {
+		String path = System.getProperty("user.dir");
+		String separator = FileSystems.getDefault().getSeparator();
+		File folder = new File(path + separator + "folder");
+		folder.mkdir();
+		assertFalse(OpenmrsUtil.folderContains(folder,"file"));
+		folder.delete();
+	}
+	
+	/**
+	 * @see OpenmrsUtil#folderContains(File, String)
+	 */
+	@Test
+	public void folderContains_shouldReturnFalseWhenGivenStringIsNull() {
+		String path = System.getProperty("user.dir");
+		String separator = FileSystems.getDefault().getSeparator();
+		File folder = new File(path + separator + "folder");
+		folder.mkdir();
```

Comment:
Don't you think your doing alot of code duplication of the above four lines of code in many methods?

Reply:
Want me to use a private method?

Reply:
Clean code should be reusable instead of 'copy & paste'. https://www.quora.com/How-to-reuse-a-code-in-Java

You may want to separate concerns by creating another method